### PR TITLE
Fix: double slashes in redirect URL

### PR DESCRIPTION
**Pull request by zhaohuabing**
This PR fix  [double splashes in the redirect URL
](https://github.com/envoyproxy/gateway/issues/2976)

When an HTTPRoute's match is a path prefix and has a redirect filter with a replacePrefixMatch "/",
A redirect routeaction with  regex_rewrite is generated to match both the prefix with and without trailing "/" to avoid double "/" in the redirect URL.

For example, given the following HTTPRoute:

```
 - matches:
   - path:
       type: PathPrefix
       value: /api/foo/
   filters:
   - requestRedirect:
       path:
         replacePrefixMatch: /
         type: ReplacePrefixMatch
     type: RequestRedirect
```

 The following route will be created:

```
              routes:
              - match:
                  path_separated_prefix: "/api/foo"
                redirect:
                  regex_rewrite:       
                    pattern:
                      regex: "^\/api\/foo\/*"
                    substitution: /
```

See https://github.com/envoyproxy/gateway/issues/2976 for more details.

Fix: https://github.com/envoyproxy/gateway/issues/2976

### CODE REVIEW

**Comment by arkodg** [comment]
hey @zhaohuabing I found some code under URLRewrite, https://github.com/envoyproxy/gateway/blob/485edbe9ac3f049d9bd3bfe9c6a36c63351ba127/internal/xds/translator/route.go#L332 , can we reuse this ?

File: internal/xds/translator/route.go [hunk 1]
```diff
@@ -303,6 +312,25 @@ func buildXdsRedirectAction(redirection *ir.Redirect) *routev3.RedirectAction {
 	return routeAction
 }
 
+// useRegexRewriteForPrefixMatchReplace checks if the regex rewrite should be used for prefix match replace
+// due to the issue with Envoy not handling the case of "//" when the replace string is "/".
+// See: https://github.com/envoyproxy/envoy/issues/26055
+func useRegexRewriteForPrefixMatchReplace(pathMatch *ir.StringMatch, prefixMatchReplace string) bool {
+	return pathMatch != nil &&
+		pathMatch.Prefix != nil &&
+		*pathMatch.Prefix != "/" &&
```
Comment (arkodg) [inline internal/xds/translator/route.go:321]: this line is new ?

**Comment by arkodg** [comment]
looks good, thanks for reusing the existing code !
minor comment, else LGTM

File: internal/xds/translator/route.go (hunk 1 above)
Comment (zhaohuabing) [reply internal/xds/translator/route.go:321]: It's an edge case. We don't need to use prefixMatchReplace if the patchMatch prefix is "/".

```
            virtual_hosts:
            - name: service  
              domains:
              - "*" 
              routes:
              - match:
                  prefix: "/"
                redirect:
                  prefixRewrite: "/"
```

**Review by arkodg — APPROVED** [review]
LGTM thanks for fixing this quirk !
and also adding e2es 🚀

**Review by zirain — APPROVED** [review]

**Pull request merged**